### PR TITLE
Improvement/24918 adapter config rework

### PR DESCRIPTION
**Pull request by Florian-Limpoeck**

### CODE REVIEW

**Review by YannickWeber — APPROVED** [review]

File: src/main/java/com/hivemq/adapter/sdk/api/config/MqttUserProperty.java [hunk 1]
```diff
@@ -97,14 +77,14 @@ public boolean equals(final @Nullable Object o) {
...
+        int result = propertyName.hashCode();
+        result = 31 * result + propertyValue.hashCode();
         return result;
     }
 
     @Override
     public @NotNull String toString() {
-        return "UserProperty{" +
+        return "MqttUserProperty{" +
                 "propertyName='" + propertyName + '\'' +
```
Comment (Florian-Limpoeck) [inline src/main/java/com/hivemq/adapter/sdk/api/config/MqttUserProperty.java:88]: maybe regenerate with fields set to "name" and "value"

File: src/main/java/com/hivemq/adapter/sdk/api/config/PollingContext.java [hunk 2]
```diff
@@ -26,12 +26,12 @@ public interface PollingContext {
...
      * @return the topic on which the data will be available
      */
-    @Nullable String getDestinationMqttTopic();
+    @NotNull String getMqttTopic();
 
     /**
      * @return the Quality-of-Service for the MQTT publish containing the data
      */
-    int getQos();
+    int getMqttQos();
```
Comment (codepitbull) [inline src/main/java/com/hivemq/adapter/sdk/api/config/PollingContext.java:34]: Could this be bound to an enum?

Comment (Florian-Limpoeck) [reply src/main/java/com/hivemq/adapter/sdk/api/config/PollingContext.java:34]: absolutely, but this PR is only about config and naming streamline and imho big enough. We can go over the SDK at another point and check the actual returnvalues, we have to anyway when we implement the other direction.

Comment (codepitbull) [reply src/main/java/com/hivemq/adapter/sdk/api/config/PollingContext.java:34]: Ok, sorry, added the same comment in the other places too.
Should I create a ticket and do that change from there?

Comment (Florian-Limpoeck) [reply src/main/java/com/hivemq/adapter/sdk/api/config/PollingContext.java:34]: that would be great yeah

**Pull request merged**